### ISSUE TITLE
Attribute error while trying to seriate data

### ISSUE THREAD

**Issue by DeepaMahm** [src-d/seriate#3]
Hi,
 I am using the following code
```
import os
import numpy
import pickle
from seriate import seriate
from scipy.spatial.distance import pdist

def serialize_data(f_input):

    if os.path.exists(f_input):
        with open(f_input, "rb") as f:
            df = pickle.load(f)
            pprint(df.head())
            elements = df.values
            pprint(elements)
            print(seriate(pdist(elements)))


if __name__ == '__main__':
    f_input = #input_file,
    serialize_data(f_input)
```
I get the following error while trying to seriate data using [this input file](https://github.com/DeepaMahm/cytoscape/blob/master/corr_data.pkl),
```
index = assignment.Value(routing.NextVar(index))
AttributeError: 'NoneType' object has no attribute 'Value'
```
I am not sure what's causing the error, I could use the same input and plot a heatmap.
Could you please look into this?

**Comment by vmarkovtsev** [comment]
This is likely to be an updated ortools version which breaks the code. @Guillemdb should figure this out. Meanwhile, can you please try to downgrade ortools to 6.9.5824

**Comment by DeepaMahm** [comment]
@vmarkovtsev Thanks a lot for the response. Downgrading ortools to 6.9.5824 helped to seriate
the input matrix of 1800 rows x 1800 columns (prior to the downgrade, the same input was causing the error that was reported). 

However, when the matrix size is increased from 1800 x 1800 to 5700 x 5700 the above-mentioned error occurs again.
Any suggestions? 

@Guillemdb Could you please look into this?

**Comment by Guillemdb** [comment]
I have been looking into it. I am not 100% sure but it may be due to the algorithm timing out before finding a valid solution. 

Could you please try your code again selecting a timeout bigger than the default 2s?

This code worked for me in the latest ortools version with the data you initially provided, and it failed when used with the default timeout.
```python
import os
import numpy
import pickle
from seriate import seriate
from scipy.spatial.distance import pdist

def serialize_data(f_input):

    if os.path.exists(f_input):
        with open(f_input, "rb") as f:
            df = pickle.load(f)
            print(df.head())
            elements = df.values
            print(elements)
            print(seriate(pdist(elements), timeout=10))

if __name__ == '__main__':
    f_input = "corr_data.pkl"
    serialize_data(f_input)
```

**Comment by Guillemdb** [comment]
@vmarkovtsev This probably can be solved by specifying an initial solution strategy:

```python
from ortools.constraint_solver import routing_enums_pb2
search_parameters = pywrapcp.DefaultRoutingSearchParameters()
search_parameters.first_solution_strategy = routing_enums_pb2.FirstSolutionStrategy.PATH_CHEAPEST_ARC
```
I will try this solution myself this evening, and in case it does not work I suggest increasing the default timeout value and issue a warning when no solution is found.

**Comment by DeepaMahm** [comment]
@Guillemdb Increasing the default timeout value helps. Thanks a lot. Using the output that is returned, should the rows (or columns) of the correlation matrix be reordered and used as the input for plotting heatmap?

**Comment by Guillemdb** [comment]
@DeepaMahm Im glad I could help you :) 

Yes, I use seriate output that way as indexes to sort the rows, but as a correlation matrix is symmetrical it should also work on the columns.

**Comment by vmarkovtsev** [comment]
This means that the algorithm's performance became worse. It sucks.

However, I think we should address this problem on our side. @Guillemdb may I ask you to implement the following:

1. If we get `None`-s in the solution, raise an `IncompleteSolutionError` which suggests to increase the timeout or set it to 0.
2. If the user specifies a 0 timeout: start with 1 second, if there are `None`-s, increase the timeout 2 times, repeat until a complete solution is found.

**Comment by DeepaMahm** [comment]
@Guillemdb  Thanks a lot for the response. I tried reordering the rows using the output obtained after seriate step.

```
import os
import numpy
import pickle
import plotly.graph_objs as go
from plotly.offline import plot
from pprint import pprint
from seriate import seriate
from scipy.spatial.distance import pdist


def serialize_data(f_input):

    if os.path.exists(f_input):
        with open(f_input, "rb") as f:
            # prior to seriation
            df = pickle.load(f)
            pprint(df.head())

            corr_data = [go.Heatmap(z=df.values.tolist(), colorscale='Viridis')]
            plot(corr_data, filename= 'plot1.html')

            # seriation
            elements = df.values  #numpy array
            idx = seriate(pdist(elements), timeout=50)
            df = df.reset_index()
            df_s = df.reindex(idx)
            corr_s_data = [go.Heatmap(z=df_s.values.tolist(), colorscale='Viridis')]
            plot(corr_s_data, filename='seriation.html')


if __name__ == '__main__':
    f_input = #input file
    serialize_data(f_input)


```
I expected the yellow, green and blue to be sorted. But it appears more or less like the plot without seriation. ([plot after seriation](https://github.com/DeepaMahm/cytoscape/blob/master/seriation.png)). I am not sure if I have missed something.

**Comment by Guillemdb** [comment]
Are you sure that Plotly.HeatMap respects the input order of the data?

 Could you please sort the distance matrix using the indexes from seriate and then use`plt.imshow()`/`plt.matshow` for visualizing the matrix. That plot would help me a lot to understand what is going on.

I know that `holoviews.heatmap()` does not allows you to visualize seriated data. It is possible that if you want interactivity you will need to do some hacking.

**Comment by DeepaMahm** [comment]
@Guillemdb 

> Could you please sort the distance matrix using the indexes from seriate and then use`plt.imshow()`/`plt.matshow` for visualizing the matrix. That plot would help me a lot to understand what is going on.
> I assume you meant sorting the correlation matrix (or should I convert the correlation matrix into a distance matrix) that is stored in `df` , `df` contains the input.

I am sorting `df` using the indexes from seriate() in the following lines of code

```
idx = seriate(pdist(elements), timeout=50)
df = df.reset_index()
df_s = df.reindex(idx)

```
Is this ok?

I will plot the contents in df_s using `plt.imshow()` / `plt.matshow` and try again. I am not sure if plotly preserves the order of input.

**Comment by vmarkovtsev** [comment]
The original issue has been fixed by #5
